### PR TITLE
Morgue stays create|edit|delete

### PR DESCRIPTION
**Pull request by Tatemsa**
[3](https://github.com/uni2growcm/u2g-openhospital-gui/issues/3)

### CODE REVIEW

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        if (jNewButton == null) {
+            jNewButton = new JButton();
+            jNewButton = new JButton(MessageBundle.getMessage("angal.common.new.btn"));
+            jNewButton.setMnemonic(MessageBundle.getMnemonic("angal.common.new.btn.key"));
+            jNewButton.addActionListener(actionEvent -> {
+                    mortuaryStay = new MortuaryStay("","", "",0,0);
+                    MortuaryStaysEdit newrecord = new MortuaryStaysEdit(myFrame, mortuaryStay, true);
+                    newrecord.addMortuaryStaysListener(MortuaryStaysBrowser.this);
+                    newrecord.setVisible(true);
+                });
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java]: fix indentation here

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+
+    /**
+     * This is the default constructor; we pass the parent frame
+     * (because it is a jdialog), the arraylist and the selected
+     * row because we need to update them
+     */
+    public MortuaryStaysEdit(JFrame parent, MortuaryStay old, boolean inserting) {
+        super(parent, true);
+        insert = inserting;
+        mortuaryStay = old;        //operation will be used for every operation
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:122]:
```suggestion
        mortuaryStay = old;
```n
```

Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
        mortuaryStay = old;
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+     *
+     * @return javax.swing.JButton
+     */
+    private JButton getOkButton() {
+        if (okButton == null) {
+            okButton = new JButton(MessageBundle.getMessage("angal.common.ok.btn"));
+            okButton.setMnemonic(MessageBundle.getMnemonic("angal.common.ok.btn.key"));
+
+            okButton.addActionListener(actionEvent -> {
+                if(insert){
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                if (insert) {
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+                            MessageDialog.error(this, "angal.mortuarystays.codealreadyinuse.msg");
+                            return;
+                        }
+                    } catch (OHServiceException e) {
+                        throw new RuntimeException(e);
+                    }
+
+                }
+                name = nameTextField.getText().trim();
+                if(name.isEmpty()){
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                if (name.isEmpty()) {
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+                    maxD = Integer.parseInt(maxDTextField.getText());
+                } catch (NumberFormatException f) {
+                    MessageDialog.error(this, "angal.mortuarystays.insertavalidmaxdnumber.msg");
+                    return;
+                }
+                if (maxD < 0) {
+                    MessageDialog.error(this, "angal.mortuarystays.insertavalidmaxdnumber.msg");
+                    return;
+                }
+                if(minD >= maxD){
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                if (minD >= maxD) {
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+                        OHServiceExceptionUtil.showMessages(ex);
+                    }
+                    if (result) {
+                        fireMortuaryStaysUpdated();
+                    }
+                }
+                if (!result) {
+                    MessageDialog.error(null, "angal.common.datacouldnotbesaved.msg");
+                }
+                else {
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]: else shoul be after }

File: src/main/java/org/isf/medicals/gui/MedicalEdit.java [hunk 7]
```diff
@@ -400,7 +422,52 @@ private VoLimitedTextField getCodeTextField() {
...
+	 * This method initializes dosingTextField
+	 * 
+	 * @return javax.swing.JTextField
+	 */
+	private VoLimitedTextField getDosingTextField() {
+		if (dosingTextField == null) {
+			dosingTextField = new VoLimitedTextField(50);
+			if (medical != null && medical.getDosing() != null ) {
+				dosingTextField.setText(medical.getDosing());
+				}
```
Comment (JantBogard) [inline src/main/java/org/isf/medicals/gui/MedicalEdit.java]:
```suggestion
			}
```

**Review by JantBogard — CHANGES_REQUESTED** [review]

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java [hunk 8]
```diff
@@ -51,9 +52,27 @@
  * @author Rick
  *
  */
-public class MortuaryStaysBrowser extends ModalJFrame {
+public class MortuaryStaysBrowser extends ModalJFrame implements MortuaryStaysListener{
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java]:
```suggestion
public class MortuaryStaysBrowser extends ModalJFrame implements MortuaryStaysListener {
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 9]
```diff
@@ -0,0 +1,461 @@
...
+     *
+     * @return javax.swing.JPanel
+     */
+    private JPanel getDataPanel() {
+        if (dataPanel == null) {
+            dataPanel = new JPanel();
+            GridBagLayout gblDataPanel = new GridBagLayout();
+            gblDataPanel.columnWeights = new double[] { 0.0, 1.0 };
+            dataPanel.setLayout(gblDataPanel);
+            JLabel codeLabel = new JLabel(MessageBundle.getMessage("angal.common.codestar"));
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:167]: please use key for message bundle don't add existing key use common key for common element please do it on all the file

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 10]
```diff
@@ -0,0 +1,461 @@
...
+                        return;
+                    }
+
+                    try {
+                        if (mortuaryStayManager.isCodePresent(code)) {
+                            MessageDialog.error(this, "angal.mortuarystays.codealreadyinuse.msg");
+                            return;
+                        }
+                    } catch (OHServiceException e) {
+                        throw new RuntimeException(e);
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                        OHServiceExceptionUtil.showMessages(e);;
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 11]
```diff
@@ -0,0 +1,461 @@
...
+
+    /**
+     * This method initializes buttonPanel
+     *
+     * @return javax.swing.JPanel
+     */
+    private JPanel getButtonPanel() {
+        if (buttonPanel == null) {
+            buttonPanel = new JPanel();
+            buttonPanel.add(getOkButton(), null);
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
            buttonPanel.add(getOkButton());
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 12]
```diff
@@ -0,0 +1,461 @@
...
+            buttonPanel.add(getCancelButton(), null);
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
            buttonPanel.add(getCancelButton());
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 13]
```diff
@@ -0,0 +1,461 @@
...
+
+    public void removeMortuaryStaysListener(MortuaryStaysListener listener) {
+        mortuaryStaysListeners.add(MortuaryStaysListener.class, listener);
+    }
+
+    private void fireMortuaryStaysInserted() {
+        AWTEvent event = new AWTEvent(new Object(), AWTEvent.RESERVED_ID_MAX + 1) {
+
+            private static final long serialVersionUID = 1L;
+        };
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:73]:
```suggestion
        }
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 14]
```diff
@@ -0,0 +1,461 @@
...
+        for (EventListener listener : listeners) {
+            ((MortuaryStaysEdit.MortuaryStaysListener) listener).mortuaryStaysInserted(event);
+        }
+    }
+
+    private void fireMortuaryStaysUpdated() {
+        AWTEvent event = new AWTEvent(new Object(), AWTEvent.RESERVED_ID_MAX + 1) {
+
+            private static final long serialVersionUID = 1L;
+        };
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:85]:
```suggestion
        }
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 15]
```diff
@@ -0,0 +1,461 @@
...
+                    if (result) {
+                        fireMortuaryStaysInserted();
+                    }
+                } else {
+                    try { // updating
+                        savedMortuaryStay = mortuaryStayManager.update(mortuaryStay);
+                        if (savedMortuaryStay != null) {
+                            result = true;
+                        }
+                    } catch (OHServiceException ex) {
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                    } catch (OHServiceException e) {
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java [hunk 16]
```diff
@@ -0,0 +1,461 @@
...
+                        OHServiceExceptionUtil.showMessages(ex);
```
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java]:
```suggestion
                        OHServiceExceptionUtil.showMessages(e);
```

File: bundle/language_en.properties [hunk 17]
```diff
@@ -1151,7 +1152,7 @@ angal.mortuarystays.deletemortuarystays.fmt.msg
 angal.mortuarystays.dmax.col                                                                           = MAX DAYS
 angal.mortuarystays.dmin.col                                                                           = MIN DAYS
 angal.mortuarystays.editmortuarystays.title                                                            = Edit Mortuary
-angal.mortuatytstays.insertavalidmaxdnumber.msg                                                        = Insert a valid maximun days number
+angal.mortuarystays.insertavalidmaxdnumber.msg                                                        = Insert a valid maximun days number
```
Comment (ArnaudFofou) [inline bundle/language_en.properties]:
```suggestion
angal.mortuarystays.insertavalidmaxdnumber.msg                                                         = Insert a valid maximun days number
```

File: bundle/language_fr.properties [hunk 18]
```diff
@@ -1141,7 +1142,7 @@ angal.mortuarystays.deletemortuarystays.fmt.msg
 angal.mortuarystays.dmax.col                                                                           = MAX DAYS
 angal.mortuarystays.dmin.col                                                                           = JOURS MIN
 angal.mortuarystays.editmortuarystays.title                                                            = Editer la morgue
-angal.mortuatytstays.insertavalidmaxdnumber.msg                                                        = Ins�rer un nombre de jours maximum valide
+angal.mortuarystays.insertavalidmaxdnumber.msg                                                        = Ins�rer un nombre de jours maximum valide
```
Comment (ArnaudFofou) [inline bundle/language_fr.properties]:
```suggestion
angal.mortuarystays.insertavalidmaxdnumber.msg                                                         = Ins�rer un nombre de jours maximum valide
```

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java (hunk 13 above)
Comment (Tatemsa) [reply src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:73]: This is an assignment instruction

Comment (Tatemsa) [reply src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:73]: This is an assignment instruction, and must end with a semicolon

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java (hunk 14 above)
Comment (Tatemsa) [reply src/main/java/org/isf/mortuarystays/gui/MortuaryStaysEdit.java:85]: This is an assignment instruction, and must end with a semicolon

File: src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java (hunk 8 above)
Comment (ArnaudFofou) [inline src/main/java/org/isf/mortuarystays/gui/MortuaryStaysBrowser.java]:
```suggestion
public class MortuaryStaysBrowser extends ModalJFrame implements MortuaryStaysListener {
```

File: bundle/language_en.properties [hunk 19]
```diff
@@ -1151,7 +1152,7 @@ angal.mortuarystays.deletemortuarystays.fmt.msg
 angal.mortuarystays.dmax.col                                                                           = MAX DAYS
 angal.mortuarystays.dmin.col                                                                           = MIN DAYS
 angal.mortuarystays.editmortuarystays.title                                                            = Edit Mortuary
-angal.mortuatytstays.insertavalidmaxdnumber.msg                                                        = Insert a valid maximun days number
+angal.mortuarystays.insertavalidmaxdnumber.msg                                                         = Insert a valid maximun days number
```
Comment (ArnaudFofou) [inline bundle/language_en.properties:1156]:
```suggestion
angal.mortuarystays.insertavalidmaxdnumber.msg                                                         = Insert a valid maximun days number
```

**Pull request closed without merging**